src/Rules/IRuleMetadataProvider.cs

```diff
@@ -39,34 +42,55 @@ public interface IRuleMetadataProvider
     [PartCreationPolicy(CreationPolicy.Shared)]
     public class RuleMetadataProvider : IRuleMetadataProvider
     {
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        public RuleMetadataProvider(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
         public IRuleHelp GetRuleHelp(Language language, string ruleKey)
         {
-            string description = language?.ServerLanguage?.Key == null 
+            string description = language?.ServerLanguage?.Key == null
                 ? Resources.Rules_DescriptionForMissingRule
                 : GetDescription(language, ruleKey);
 
             return new RuleHelp(language, ruleKey, description);
         }
 
         private string GetDescription(Language language, string ruleKey)
+        {
+            var ruleInfo = LoadRuleInfo(language, ruleKey);
+            return ruleInfo?.Description ?? Resources.Rules_DescriptionForMissingRule;
+        }
+        
+        private IRuleInfo LoadRuleInfo(Language language, string ruleKey)
         {
             var resourcePath = CalcFullResourceName(language, ruleKey);
-            using (var stream = GetType().Assembly.GetManifestResourceStream(resourcePath))
+            try
             {
-                if (stream != null)
+                using (var stream = GetType().Assembly.GetManifestResourceStream(resourcePath))
                 {
-                    using (var reader = new StreamReader(stream))
+                    if (stream != null)
                     {
-                        return reader.ReadToEnd();
+                        using (var reader = new StreamReader(stream))
+                        {
+                            var data = reader.ReadToEnd();
+                            return JsonConvert.DeserializeObject<RuleInfo>(data);
+                        }
                     }
                 }
             }
-
-            return Resources.Rules_DescriptionForMissingRule;
+            catch (Exception ex)
+            {
+                logger.LogVerbose(Resources.MetadataProvider_ErrorLoadingJson, resourcePath, ex);
```

Comment:
Difficult to test this bit...

Reply:
Hmm yea I can't think of any simple ways either :/